src/animate/AnimatorTimeline.js

```diff
@@ -37,21 +40,29 @@ class AnimatorTimeline {
      * @private
      */
     destroy() {
-        this.init(null, null, false, 0, 0, null);
+        this.instance.beforeUpdateTimeline = null;
+        this.init(null, 0, 0, false, null);
         AnimatorTimeline._pool.push(this);
     }
 
     /**
      * Is the animation complete
-     * @method update
+     * @method before
      * @return {Boolean} 
      * @private
      */
-    update() {
-        if (this.instance.currentFrame >= this.end) {
+    update(instance) {
+        if (instance.currentFrame >= this.end) {
+
+            // In case we over-shoot the current frame becuase of low FPS
+            instance.currentFrame = this.end;
+
             if (this.loop) {
-                this.instance.gotoAndPlay(this.start);
+                instance.gotoAndPlay(this.start);
```

Comment:
The current frame is already being set above. would replace this gotoAndPlay call with:

```js
instance._updateTimeline();
instance.gotoAndPlay(this.start);
```